src/internal/components/QrCode/useMatrix.ts

```diff
@@ -0,0 +1,27 @@
+import QRCode from 'qrcode';
+import { useMemo } from 'react';
+
+export function useMatrix(
+  value: string,
+  errorCorrectionLevel: 'L' | 'M' | 'Q' | 'H',
+) {
+  if (!value) {
+    return [];
+  }
+
+  const matrix = useMemo(() => {
```

Comment:
can't conditionally call useMemo https://react.dev/reference/rules/rules-of-hooks